src/yunohost/app.py

```diff
@@ -156,10 +156,23 @@ def app_info(app, full=False):
 
 
 def _app_upgradable(app_infos):
+    from packaging import version
 
     # Determine upgradability
     # In case there is neither update_time nor install_time, we assume the app can/has to be upgraded
 
+    # Firstly use the version to know if an upgrade is available
+    app_is_in_catalog = bool(app_infos.get("from_catalog"))
+    upgrade_only_if_version_changes = app_infos["manifest"].get('integration', {}).get("upgrade_only_if_version_changes", None) is True
+    installed_version = version.parse(app_infos["version"])
+    version_in_catalog = version.parse(app_infos.get("from_catalog", {}).get("manifest", {}).get("version", "0~ynh0"))
+    
+    if app_is_in_catalog and '~ynh' in app_infos["version"]:
```

Comment:
```suggestion
    if app_is_in_catalog and '~ynh' in str(installed_version) and '~ynh' in str(version_in_catalog):
```

Reply:
(or maybe I missed some reason to not check the `version_in_catalog` here as well ?)

Reply:
I thought that it's enought to check on on one side because if the official version is used on one side it will probably be the same on the other side. But yes it's more robust to check on booth side.